tests/misc/lang_mapper.py

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from collections import OrderedDict
+
+import pytest
+
+from pytest_pootle.factories import ProjectDBFactory
+
+from pootle.core.delegate import config, lang_mapper
+from pootle_config.utils import ObjectConfig
+from pootle_language.models import Language
+
+
+CUSTOM_LANG_MAP_PRESETS = OrderedDict(
+    [["foo",
+      [["testfoo1", "en"],
+       ["testfoo2", ""],
+       ["testfoo3", "language0"],
+       ["testfoo4", "foo"]]],
+     ["bar",
+      [["testbar1", "en"],
+       ["testbar2", "language1"],
+       ["testbar3", "language0"],
+       ["testbar4", "bar"]]],
+     ["baz",
+      [["testbaz1", "en"],
+       ["testbaz2", "language1"],
+       ["testbaz3", "language0"],
+       ["testbaz4", "baz"]]]])
```

Comment:
foo, bar, baz makes it frustratingly hard to understand what exactly is being tested.  Real codes for real languages would have helped way way more.